specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

```diff
@@ -21156,6 +22197,228 @@
+          {
+            "value": "Suspending"
+          },
+          {
+            "value": "Reinstating"
+          },
+          {
+            "value": "Online"
+          },
+          {
+            "value": "Suspended"
+          },
+          {
+            "value": "CreationFailed"
+          },
+          {
+            "value": "DeletionFailed"
+          }
+        ]
+      }
+    },
+    "ServerlessEndpointStatus": {
+      "additionalProperties": false,
+      "properties": {
+        "metrics": {
+          "additionalProperties": {
+            "type": "string",
+            "x-nullable": true
+          },
+          "description": "The model-specific metrics from the backing inference endpoint.",
+          "readOnly": true,
+          "type": "object",
+          "x-ms-mutability": [
+            "read"
+          ],
+          "x-nullable": true
+        }
+      },
+      "type": "object"
+    },
+    "ServerlessEndpointTrackedResource": {
+      "additionalProperties": false,
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/ManagedServiceIdentity",
+          "description": "Managed service identity (system assigned and/or user assigned identities)"
+        },
+        "kind": {
+          "description": "Metadata used by portal/tooling/etc to render different UX experiences for resources of the same type.",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/ServerlessEndpoint",
+          "description": "[Required] Additional attributes of the entity."
+        },
+        "sku": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Sku",
+          "description": "Sku details required for ARM contract for Autoscaling."
+        }
+      },
+      "required": [
+        "location",
+        "properties"
+      ],
+      "type": "object",
+      "x-ms-client-name": "ServerlessEndpoint"
+    },
+    "ServerlessEndpointTrackedResourceArmPaginatedResult": {
+      "additionalProperties": false,
+      "description": "A paginated list of ServerlessEndpoint entities.",
+      "properties": {
+        "nextLink": {
+          "description": "The link to the next page of ServerlessEndpoint objects. If null, there are no additional pages.",
+          "type": "string"
+        },
+        "value": {
+          "description": "An array of objects of type ServerlessEndpoint.",
+          "items": {
+            "$ref": "#/definitions/ServerlessEndpointTrackedResource"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "ServerlessInferenceEndpoint": {
+      "additionalProperties": false,
+      "properties": {
+        "headers": {
```

Comment:
>headers [](http://example.com/codeflow?start=9&length=7)

Similar consideration here regarding additionalProperties pattern.

Reply:
the headers property here is meant to describe a set of request headers that the user must pass along in their inferencing API request. for that reason this needs to be represented as an additionalProperties

Reply:
added a suppression rule for this via https://github.com/Azure/azure-rest-api-specs/pull/29041/commits/69f22f64d8f73406010962ff88214a9cd5e0f0ca and https://github.com/Azure/azure-rest-api-specs/pull/29041/commits/e9de06da929f8a4456ccd8b4ec5cbed13ab6a23f

the suppression was extended to the dependent definitions because LintDiff seemed to be failing those dependencies even when the root property with additionalProperties was suppressed